lib/auth/init.go

```diff
@@ -611,15 +616,30 @@ type PresetRoleManager interface {
 	UpsertRole(ctx context.Context, role types.Role) error
 }
 
-// createPresets creates preset resources (eg, roles).
-func createPresets(ctx context.Context, rm PresetRoleManager) error {
+// createPresetRoles creates preset role resources
+func createPresetRoles(ctx context.Context, rm PresetRoleManager) error {
 	roles := []types.Role{
 		services.NewPresetGroupAccessRole(),
 		services.NewPresetEditorRole(),
 		services.NewPresetAccessRole(),
 		services.NewPresetAuditorRole(),
+		services.NewPresetAutomaticAccessApproverRole(),
 	}
 	for _, role := range roles {
+		// Some roles are only valid for enterprise Teleport, and so will be
+		// nil for an OSS build and can be skipped
+		if role == nil {
+			continue
+		}
+
+		if types.IsSystemResource(role) {
+			// System resources * always* get reset on every auth startup
+			if err := rm.UpsertRole(ctx, role); err != nil {
+				return trace.Wrap(err, "failed upserting system role %s", role.GetName())
+			}
```

Comment:
This will generate audit events for each Auth startup
I'm not sure we want this 🤔 

Related PR
https://github.com/gravitational/teleport/pull/20824

Reply:
This opens up the question of defining equality for `Role`s and `User`s, which I'm not sure I want to tackle as part of this PR.

I've created ticket #28183 to capture this.